helm/alfresco-identity-service/values.yaml

```diff
@@ -58,7 +53,114 @@ realm:
         groups:
           - "/admin"
           - "/testgroup"
-
+extraClients:
+  alfresco:
+    clientRoles:
+      - name: test_client_role1
+        scopeParamRequired: false
+        composite: false
+        clientRole: true
+      - name: test_client_role2
+        scopeParamRequired: false
+        composite: false
+        clientRole: true
+    id: be13165d-2792-43aa-abdb-2c6cdb627184
+    clientId: alfresco
+    surrogateAuthRequired: false
+    enabled: true
+    clientAuthenticatorType: client-secret
+    secret: "**********"
+    redirectUris:
+      - "*"
```

Comment:
@ryandawsonuk Wildcarding the redirectUris and webOrigins would be less secure.  Is this a practice we would want to advocate?

Reply:
I would be happy to change this to localhost:* as it was - it seems to me it doesn't matter as it would just be an example and not really part of the base realm. Consumers would define it themselves. [APS for example already does](https://git.alfresco.com/process-services-public/alfresco-process-infrastructure-deployment/blob/develop/helm/alfresco-process-infrastructure/alfresco-aps-realm.json#L514). 
So I was suggesting that README section could be removed because the client specification wouldn't be part of the base realm. AIS would provide common base configurations and consumers would build on top of that and add their own clients, since only they know which apps in their stack are clients. Currently consumers are [just overriding the entire realm and not retaining any of it](https://git.alfresco.com/process-services-public/alfresco-process-infrastructure-deployment/blob/develop/helm/alfresco-process-infrastructure/templates/realm-secret.yaml#L12) and I believe [this is because the flexibility to define clients is needed](https://git.alfresco.com/process-services-public/alfresco-process-infrastructure-deployment/blob/develop/helm/alfresco-process-infrastructure/alfresco-aps-realm.json#L509).

Reply:
All good points and I think as far as usability and testing go for a base it may be ok. Do you have a link to the documentation/README for APS that would discuss overriding these so we can use common vocabulary? Or is this something that needs to be developed still?

Reply:
> it doesn't matter as it would just be an example and not really part of the base realm. Consumers would define it themselves. APS for example currently does.

We shouldn't be encouraging wide open redirectUris and webOrigins.

> the client specification wouldn't be part of the base realm.

As mentioned in my review, I believe the `alfresco` client should remain part of the base realm.

> AIS would provide common base configurations and consumers would build on top of that and add their own clients, since only they know which apps in their stack are clients.

Agreed.

> Currently consumers are just overriding the entire realm and not retaining any of it and I believe this is because the flexibility to define clients is needed.

Yes, we're aware and that's unfortunate, and I'm not sure why additional clients are needed in that particular case, but if the `extraClients` portion of this PR is merged that will be configurable.

> The idea is to get to setup where activiti and APS could both use in the same way and build on a common base.

Absolutely.